### PR TITLE
chore: Add benchmark test for long running subscription

### PR DESCRIPTION
**Pull request by nrbnlulu**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

*

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the CONTRIBUTING document.
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

## Summary by Sourcery

Add a benchmark test for long-running subscriptions to assess performance and stability during extended execution.

Tests:
- Add a benchmark test for long-running subscriptions to evaluate performance over extended periods.

### CODE REVIEW

File: tests/benchmarks/test_subscriptions.py [hunk 1]
```diff
@@ -24,3 +26,22 @@ async def _run():
...
 
     benchmark(lambda: asyncio.run(_run()))
+
+
+@pytest.mark.benchmark
+def test_subscription_long_run(benchmark: BenchmarkFixture) -> None:
+    s = """
+    subscription {
+        longRunning
+    }
```
Comment (patrick91) [inline tests/benchmarks/test_subscriptions.py:36]: sorry to be pedantic, but could we make the subscription take a param? so we can test with 100 and 10_000 in one single test? :D

Comment (nrbnlulu) [reply tests/benchmarks/test_subscriptions.py:36]: np. though I don't think I understand wym in a single tests?

Comment (patrick91) [reply tests/benchmarks/test_subscriptions.py:36]: similar to this: https://github.com/strawberry-graphql/strawberry/blob/main/tests/benchmarks/test_complex_schema.py#L89-L101

**Comment by nrbnlulu** [comment]
I haven't touched typecheckers why are they failing?

File: tests/benchmarks/test_subscriptions.py [hunk 2]
```diff
@@ -24,3 +26,25 @@ async def _run():
             assert value.data["something"] == "Hello World!"
 
     benchmark(lambda: asyncio.run(_run()))
```
Comment (patrick91) [inline tests/benchmarks/test_subscriptions.py:28]: do you think we still need this test now? 🤔

Comment (nrbnlulu) [reply tests/benchmarks/test_subscriptions.py:28]: maybe. This one tests the whole lifespan of a subscription while the other is for the subscription itself.

Comment (patrick91) [reply tests/benchmarks/test_subscriptions.py:28]: let's do that!

I fixed the type issue, it's on main :D

**Review by patrick91 — APPROVED** [review]

**Pull request merged**